### PR TITLE
[docs] Clarify torch.compile module method scope

### LINKED ISSUE #141616: Clarify that `compile(module)` only affects the `forward` method.
### 📚 The doc issue

The following code silently does not trigger compilation, apparently because `forward` is not called. One can see this by using a debug flag or passing an illegal configuration like `backend="cudagraphs", mode="reduce-overhead"` to `compile`. 

Certain models like Normalizing Flows and distribution-like modules might be used without calling `forward`. 

```python
import os
os.environ["TORCH_LOGS"] = "dynamo"
import torch
from torch import Tensor, nn

class Flow(nn.Module):
    def forward(self, x: Tensor) -> Tensor:
        return x + 1.0

    def inverse(self, x: Tensor) -> Tensor:
        return x - 1.0

m = Flow().to(device="cuda")
optimized: Flow = torch.compile(m)    # type: ignore

x = torch.rand(1, 2, 3, 4).to(device="cuda")
y1 = m.inverse(x)
y2 = optimized.inverse(x)
assert torch.equal(y1, y2)
print("No compilation until here.")
optimized(x)  # <- triggers compilation of `forward`
```

### Suggest a potential alternative/fix

Clarify that when using `torch.compile(module)`, it is only triggered upon calling `forward`, and that only `forward` gets compiled. The current documentation is imprecise:

#### https://pytorch.org/tutorials/intermediate/torch_compile_tutorial.html

> In the same fashion, when compiling a module all sub-modules and methods within it, that are not in a skip list, are also compiled.

- Apparently, not "all" methods within the module get compiled, but only the ones `forward` depends on.
- Secondly, assume `MyModel.forward` calls `MyModel.helper_fn`. After compilation, if I call `helper_fn` in a standalone manner, do I get a compiled version? Or does it only compile it in an "inlined" manner?

> When you use `torch.compile`, the compiler will try to recursively compile every function call inside the target function or module inside the target function or module that is not in a skip list (such as built-ins, some functions in the torch.* namespace).

This should probably read *"inside the target module's forward method"* instead.

#### https://pytorch.org/docs/stable/generated/torch.compile.html

This should also mention that compilation only applies to forward.

### PR DESCRIPTION
**Pull request by arhxam**
Fixes #141616.

Clarify the compilation boundary when `torch.compile` receives an `nn.Module`.
Compilation is installed on the module call path that invokes `forward`; direct
calls to other methods remain eager unless those methods are compiled
separately. Code reached from `forward` can still be captured as part of that
compiled region.

The documented behavior was verified with a counting backend for both
`torch.compile(module)` and in-place `module.compile()`.

BC-breaking? No. Documentation only.

> **AI assistance disclosure:** Codex helped prepare the change, verification,
> and this PR summary.

**Human commentary:** I reviewed and verified this contribution end to end before authorizing submission.

### CODE REVIEW

**Comment by arhxam** [comment]
@williamwen42 This has passed EasyCLA and is ready for review. Could you take a look when available?
